noxfile.py

```diff
@@ -36,6 +36,20 @@ def tests_impl(
     elif sys.platform == "win32":
         memray_supported = False
 
+    # Environment variables being passed to the pytest run.
+    pytest_session_envvars = {
+        "PYTHONWARNINGS": "always::DeprecationWarning",
+    }
+
+    # In coverage 7.4.0 we can only set the setting for Python 3.12+
+    # Future versions of coverage will use sys.monitoring based on availability.
+    if (
+        isinstance(session.python, str)
```

Comment:
In what case is this not a string?

Reply:
This is mostly to make type hints happy, apparently the type of `session.python` is `True | None | str`.